packages/client/screens/MyProfile/MyFollowing/index.tsx

```diff
@@ -0,0 +1,161 @@
+                                    isCreator: false,
+                                    isDelegationOwner: false,
+                                    isMe: false,
+                                    nickname,
+                                    userID,
+                                }}
+                                disabled={false}
+                                onPress={() => {
+                                    navigation.navigate('UserProfile', {
+                                        screen: 'UserProfileIndex',
+                                        params: {
+                                            userID,
+                                        },
+                                    });
+                                }}
+                            />
+                        </View>
+                    );
+                }}
+                keyExtractor={({ userID }) => userID}
+                ListEmptyComponent={() => {
+                    return (
+                        <Text sx={{ color: 'white' }}>
+                            This user doesnot have any following
+                        </Text>
+                    );
+                }}
+                ListFooterComponent={
+                    hasMoreUsersToFetch === true
+                        ? () => {
+                              return (
+                                  <View
+                                      sx={{
+                                          flexDirection: 'row',
+                                          justifyContent: 'center',
+                                          alignItems: 'center',
+                                      }}
+                                  >
+                                      <TouchableOpacity
+                                          onPress={handleLoadMore}
+                                          style={sx({
+                                              borderRadius: 'full',
+                                              borderWidth: 2,
+                                              borderColor: 'secondary',
+                                              paddingX: 'l',
+                                              paddingY: 's',
+                                          })}
+                                      >
+                                          <Text
+                                              sx={{
+                                                  color: 'secondary',
+                                                  fontWeight: 'bold',
+                                              }}
+                                          >
+                                              Load more
+                                          </Text>
+                                      </TouchableOpacity>
+                                  </View>
+                              );
+                          }
+                        : undefined
+                }
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I recently discovered that `ListFooterComponent` prop accepts either a React function (as here) or a React component. The advantage of using a React component here is that React would be able to cache the Footer component. As it is written currently, a new component is created for each render, and I think that React is unmounting it for each render. This is an interesting thing that I want to deepen 👀.

See [React Native documentation for FlatList as reference](https://reactnative.dev/docs/flatlist#listfootercomponent)

Reply:
Interesting :open_mouth: